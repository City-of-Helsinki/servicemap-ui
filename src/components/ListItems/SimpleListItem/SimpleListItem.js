import React from 'react';
import PropTypes from 'prop-types';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import { Typography, Divider } from '@material-ui/core';

const SimpleListItem = (props) => {
  const {
    button, text, classes, link, icon, handleItemClick, role, divider, selected, srText, className,
  } = props;
  return (
    <React.Fragment>
      <ListItem
        className={className}
        button={!!link || button}
        role={link ? 'link' : role}
        component="li"
        onClick={button || link ? handleItemClick : null}
        classes={{
          root: classes.listItem,
        }}
        selected={selected}
      >
        <ListItemIcon aria-hidden className={`${classes.listIcon} ${link ? classes.link : null}`}>
          {icon}
        </ListItemIcon>

        <ListItemText
          classes={{ root: classes.textContainer }}
        >
          <Typography variant="srOnly">
            {`${srText || ''} ${text}`}
          </Typography>

          <Typography
            aria-hidden
            variant="body2"
            classes={{ root: link ? classes.link : null }}
          >
            {text}
          </Typography>
        </ListItemText>
      </ListItem>
      {divider && (
        <li aria-hidden>
          <Divider className={classes.divider} />
        </li>
      )}
    </React.Fragment>
  );
};

export default SimpleListItem;

SimpleListItem.propTypes = {
  button: PropTypes.bool,
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  text: PropTypes.string.isRequired,
  srText: PropTypes.string,
  link: PropTypes.bool,
  icon: PropTypes.objectOf(PropTypes.any),
  handleItemClick: PropTypes.func,
  role: PropTypes.string,
  divider: PropTypes.bool,
  selected: PropTypes.bool,
  className: PropTypes.string,
};

SimpleListItem.defaultProps = {
  button: false,
  srText: null,
  link: false,
  icon: null,
  handleItemClick: null,
  role: null,
  divider: false,
  selected: false,
  className: null,
};
