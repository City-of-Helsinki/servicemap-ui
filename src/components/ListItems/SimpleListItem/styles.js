export default theme => ({
  listItem: {
    minHeight: '3.5rem',
    padding: theme.spacing.unit,
  },
  textContainer: {
    padding: `${theme.spacing.unit}px 0`,
    marginLeft: theme.spacing.unitDouble,
    marginRight: theme.spacing.unitDouble,
    whiteSpace: 'pre-line',
  },
  link: {
    color: '#0000EE',
  },
  listIcon: {
    width: '1.5rem',
    height: '1.5rem',
    margin: theme.spacing.unit,
    marginRight: theme.spacing.unitDouble,
  },
  divider: {
    marginLeft: theme.spacing.unit * 9,
    marginRight: -theme.spacing.unitDouble,
  },
});
