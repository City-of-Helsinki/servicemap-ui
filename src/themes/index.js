import { createMuiTheme } from '@material-ui/core';

const focusIndicator = {
  outline: '2px solid transparent',
  boxShadow: '0 0 0 2px rgb(255, 255, 255), 0 0 0 6px rgb(71, 131, 235), 0 0 0 8px rgb(255, 255, 255)',
};

const focusIndicatorDark = {
  outline: '2px solid transparent',
  boxShadow: '0 0 0 2px rgb(255, 255, 255), 0 0 0 6px rgb(0, 0, 0), 0 0 0 8px rgb(255, 255, 255)',
};

const props = {
  // Globally disable all default mui focus effects
  MuiButtonBase: {
    disableRipple: true,
  },
  MuiButton: {
    disableRipple: true,
    disableFocusRipple: true,
  },
  MuiTypography: {
    variant: 'body2',
  },
};

const overrides = theme => ({
  MuiButtonBase: {
    root: {
      // Default keyboard focus indicator for buttons
      '&$focusVisible': theme === 'dark' ? focusIndicatorDark : focusIndicator,
    },
  },
  MuiInputBase: {
    // Default keyboard focus indicator for input fields
    root: { '&$focused': theme === 'dark' ? focusIndicatorDark : focusIndicator },
  },
  MuiListItemText: {
    root: {
      marginTop: 0,
      marginBottom: 0,
    },
  },
  MuiSvgIcon: {
    root: {
      fontSize: 24,
    },
  },
  PrivateSwitchBase: {
    root: {
      padding: 12,
    },
  },
});

const breakpoints = {
  values: {
    xs: 0,
    sm: 700,
    md: 900,
    lg: 1280,
    xl: 1920,
  },
};

const spacing = 8;

const custom = {
  body2light: {
    color: 'rgba(0,0,0,0.7)',
  },
};

const zIndex = {
  behind: -1,
  forward: 50,
  sticky: 51,
  infront: 900,
};

const typography = {
  useNextVariants: true,
  fontSize: 16,
  // Use the system font instead of the default Roboto font.
  fontFamily: [
    'Lato',
  ].join(','),
  body1: {
    fontSize: '1.043rem',
    letterSpacing: '0.03125rem', // 0.5px
    lineHeight: '1.75rem',
  },
  body2: {
    fontSize: '0.913rem',
    letterSpacing: '0.015625rem', // 0.25px
    lineHeight: '1.25rem',
  },
  subtitle1: {
    fontSize: '1.03rem',
    fontWeight: 'bold',
    letterSpacing: '0.00937rem', // 0.15px
    lineHeight: '1.5rem',
  },
  caption: {
    fontSize: '0.7725rem', // 12.36px
    fontWeight: 'bold',
    letterSpacing: '0.025rem', // 0.4px
    lineHeight: '1rem',
    color: 'rgba(0,0,0,0.6)',
  },
  h1: {
    fontSize: '6.259rem',
    letterSpacing: '-0.09375em', // -1.5px
  },
  h2: {
    fontSize: '3.862rem',
    fontWeight: 'bold',
    letterSpacing: '-0.03125rem', // -0.5px
  },
  h3: {
    fontSize: '3.089rem',
    fontWeight: 'bold',
    letterSpacing: 0,
  },
  h4: {
    fontSize: '2.214rem',
    letterSpacing: '0.015625rem', // 0.25px
  },
  h5: {
    fontSize: '1.564rem',
    letterSpacing: 0,
  },
  h6: {
    fontSize: '1.288rem',
    fontWeight: 'bold',
    letterSpacing: '0.015625rem', // 0.25px
    lineHeight: '1.5rem',
  },
  button: {
    fontSize: '0.90125rem', // 14.42px
    fontWeight: 'bold',
    letterSpacing: '0.078125rem', // 1.25px
    lineHeight: '1rem', // 16px
    color: '#ffffff',
  },
  overline: {
    fontSize: '0.7725rem', // 12.36px
    fontWeight: 'bold',
    letterSpacing: '0.125rem', // 2px
    lineHeight: '1rem',
  },
};

// Color palette for normal theme
export const paletteDefault = {
  primary: {
    main: 'rgb(25, 100, 230)',
    highContrast: '#fff',
  },
  secondary: {
    main: 'rgb(43, 47, 57)',
    hover: '#1d39ad',
  },
  background: {
    main: 'linear-gradient(340.58deg, rgb(11, 123, 237) 0%, rgb(20, 108, 232) 67.04%, rgb(25, 100, 230) 100%)',
    plain: 'rgb(20, 108, 232)',
    front: 'linear-gradient(340.58deg, rgb(11, 123, 237) 0%, rgb(20, 108, 232) 67.04%, rgb(25, 100, 230) 100%)',
  },
  white: {
    light: '#f2f2f2',
    main: '#FFFFFF',
    dark: '#949494',
    contrastText: '#000',
  },
  detail: {
    main: 'rgb(25, 100, 230)',
    alpha: 'rgba(25,100,230,0.5)',
  },
  disabled: {
    main: 'rgb(239, 239, 239)',
    strong: '#898989',
  },
  focusBorder: {
    main: 'rgb(71, 131, 235)',
  },
  warning: {
    main: '#AF0021',
  },
};

// Color palette for dark theme
export const paletteDark = {
  primary: {
    main: '#353638',
    highContrast: '#fff',
  },
  secondary: {
    main: 'rgb(43, 47, 57)',
    hover: '#5b5d61',
  },
  background: {
    main: '#4A4A4C',
    plain: '#4A4A4C',
    front: 'linear-gradient(326.21deg, rgba(0, 0, 0, 0.79) 0%, rgba(71, 71, 71, 0.79) 100%)',
  },
  white: {
    light: '#f2f2f2',
    main: '#FFFFFF',
    dark: '#949494',
    contrastText: '#000',
  },
  detail: {
    main: '#585B63',
    alpha: 'rgba(88,91,99,0.5)',
  },
  disabled: {
    main: 'rgb(239, 239, 239)',
    strong: '#898989',
  },
  warning: {
    main: '#AF0021',
  },
  focusBorder: {
    main: '#000',
  },
};

// Themes
const SMTheme = createMuiTheme({
  props,
  overrides: overrides('default'),
  breakpoints,
  typography,
  spacing,
  custom,
  palette: paletteDefault,
  zIndex,
});

const SMThemeDark = createMuiTheme({
  props,
  overrides: overrides('dark'),
  breakpoints,
  typography,
  spacing,
  custom,
  palette: paletteDark,
  zIndex,
});


export default { SMTheme, SMThemeDark };
