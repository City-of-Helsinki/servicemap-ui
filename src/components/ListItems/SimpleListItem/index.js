import SimpleListItem from './SimpleListItem';

export default SimpleListItem;
