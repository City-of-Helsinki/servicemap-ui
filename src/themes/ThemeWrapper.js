import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { ThemeProvider } from '@mui/material/styles';
import themes from '.';

const mapStateToProps = (state) => {
  const { user } = state;
  return {
    theme: user.theme,
  };
};

// Component to handle theme changes
const ThemeHandler = ({ children, theme }) => (
// Get correct theme setting from store
  <ThemeProvider theme={theme === 'dark' ? themes.SMThemeDark : themes.SMTheme}>
    {children}
  </ThemeProvider>
);
ThemeHandler.propTypes = {
  children: PropTypes.node,
  theme: PropTypes.string.isRequired,
};
ThemeHandler.defaultProps = {
  children: null,
};

const ThemeWrapper = connect(
  mapStateToProps,
)(ThemeHandler);

export default ThemeWrapper;
