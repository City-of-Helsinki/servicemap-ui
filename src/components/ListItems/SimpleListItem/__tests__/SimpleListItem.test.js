// Link.react.test.js
import React from 'react';
import { createMount } from '@material-ui/core/test-utils';
import { MuiThemeProvider } from '@material-ui/core';
import themes from '../../../../../themes';
import SimpleListItem from '../index';

// Generic required props for SimpleListItem
const mockProps = {
  text: 'Title text',
};

// eslint-disable-next-line react/prop-types
const Providers = ({ children }) => (
  <MuiThemeProvider theme={themes.SMTheme}>
    {children}
  </MuiThemeProvider>
);

describe('<SimpleListItem />', () => {
  // let render;
  let mount;

  beforeEach(() => {
    // render = createRender({ wrappingComponent: Providers });
    mount = createMount({ wrappingComponent: Providers });
  });

  afterEach(() => {
    mount.cleanUp();
  });

  const resetMount = () => {
    mount.cleanUp();
    mount = createMount({ wrappingComponent: Providers });
  };

  it('should work', () => {
    const component = mount(<SimpleListItem {...mockProps} />);
    expect(component).toMatchSnapshot();
  });

  it('simulates click event', () => {
    const mockCallBack = jest.fn();
    const component = mount(
      <SimpleListItem {...mockProps} handleItemClick={mockCallBack} button />
    );

    component.find('ListItem').simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });

  it('simulates keyboard event', () => {
    const mockCallBack = jest.fn();
    const component = mount(
      <SimpleListItem {...mockProps} handleItemClick={mockCallBack} button />
    );

    component.find('ListItem').simulate('keyDown', { which: 13 });
    component.find('ListItem').simulate('keyDown', { which: 32 });
    expect(mockCallBack.mock.calls.length).toEqual(2);
  });

  it('does show text correctly', () => {
    const component = mount(<SimpleListItem {...mockProps} />);

    const p = component.find('p');
    expect(p.at(0).text()).toEqual(mockProps.text);
  });

  it('does set select correctly', () => {
    let component = mount(<SimpleListItem {...mockProps} />);
    expect(component.find('ListItem').props().selected).toBeFalsy();

    resetMount();

    component = mount(<SimpleListItem {...mockProps} selected />);
    expect(component.find('ListItem').props().selected).toBeTruthy();
  });

  it('does set divider correctly', () => {
    let component = mount(<SimpleListItem {...mockProps} />);
    expect(component.find('Divider').exists()).toBeFalsy();

    resetMount();

    component = mount(<SimpleListItem {...mockProps} divider />);
    expect(component.find('Divider').exists()).toBeTruthy();
  });

  it('does use default accessibility attributes correctly', () => {
    const component = mount(<SimpleListItem {...mockProps} />);

    const srText = component.find('span').at(1);
    const text = component.find('p').at(0);

    // Expect screen reader texts to render correctly
    expect(component.find('ListItem').props()['aria-label'].indexOf(mockProps.text) !== -1).toBeTruthy();
    // Expect aria-hidden attributes to be placed correctly
    expect(srText.props()['aria-hidden']).toBeTruthy();
    expect(text.props()['aria-hidden']).toBeTruthy();
    // Expect role to be set
    expect(component.find('ListItem').props().role).toEqual(null);
    // Expect element to have tabIndex -1
    expect(component.find('ListItem').props().tabIndex).toEqual(-1);
  });

  it('does use given accessibility attributes correctly', () => {
    const component = mount(
      <SimpleListItem
        {...mockProps}
        role="button"
        srText="Screen reader text"
        divider
        button
      />,
    );

    const srText = component.find('span').at(1).text();
    const containsText = srText.indexOf('Screen reader text') !== -1;
    const alText = component.find('ListItem').props()['aria-label'];
    const alContainsText = alText.indexOf(mockProps.text) !== -1
    && alText.indexOf('Screen reader text') !== -1

    // Expect screen reader text to contain both text and srText values
    expect(alContainsText).toBeTruthy();
    // Expect screen reader only text to exist in separate span element
    expect(containsText).toBeTruthy();
    // Expect role to be set
    expect(component.props().role).toEqual('button');
    // Expect divider element to be hidden from screen readers
    expect(component.find('li').at(1).props()['aria-hidden']).toBeTruthy();
    // Expect element to have tabIndex 0
    expect(component.find('ListItem').props().tabIndex).toEqual(0);
  });
});
