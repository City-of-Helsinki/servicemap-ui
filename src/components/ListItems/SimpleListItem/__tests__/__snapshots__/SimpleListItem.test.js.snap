// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimpleListItem /> should work 1`] = `
<WithStyles(SimpleListItem)
  text="Title text"
>
  <SimpleListItem
    button={false}
    className={null}
    classes={
      Object {
        "divider": "SimpleListItem-divider-6",
        "itemFocus": "SimpleListItem-itemFocus-7",
        "link": "SimpleListItem-link-3",
        "listIcon": "SimpleListItem-listIcon-5",
        "listItem": "SimpleListItem-listItem-1",
        "textContainer": "SimpleListItem-textContainer-2",
        "whiteText": "SimpleListItem-whiteText-4",
      }
    }
    dark={false}
    divider={false}
    handleItemClick={null}
    icon={null}
    link={false}
    role={null}
    selected={false}
    srText={null}
    text="Title text"
  >
    <WithStyles(ForwardRef(ListItem))
      aria-label=" Title text"
      button={false}
      className={null}
      classes={
        Object {
          "root": "SimpleListItem-listItem-1",
          "selected": "SimpleListItem-itemFocus-7",
        }
      }
      component="li"
      onClick={null}
      onKeyDown={null}
      role={null}
      selected={false}
      tabIndex={-1}
    >
      <ForwardRef(ListItem)
        aria-label=" Title text"
        button={false}
        className={null}
        classes={
          Object {
            "alignItemsFlexStart": "MuiListItem-alignItemsFlexStart",
            "button": "MuiListItem-button",
            "container": "MuiListItem-container",
            "dense": "MuiListItem-dense",
            "disabled": "Mui-disabled",
            "divider": "MuiListItem-divider",
            "focusVisible": "Mui-focusVisible",
            "gutters": "MuiListItem-gutters",
            "root": "MuiListItem-root SimpleListItem-listItem-1",
            "secondaryAction": "MuiListItem-secondaryAction",
            "selected": "Mui-selected SimpleListItem-itemFocus-7",
          }
        }
        component="li"
        onClick={null}
        onKeyDown={null}
        role={null}
        selected={false}
        tabIndex={-1}
      >
        <li
          aria-label=" Title text"
          className="MuiListItem-root SimpleListItem-listItem-1 MuiListItem-gutters"
          disabled={false}
          onClick={null}
          onKeyDown={null}
          role={null}
          tabIndex={-1}
        >
          <WithStyles(ForwardRef(ListItemText))
            classes={
              Object {
                "root": "SimpleListItem-textContainer-2",
              }
            }
            key=".1"
          >
            <ForwardRef(ListItemText)
              classes={
                Object {
                  "dense": "MuiListItemText-dense",
                  "inset": "MuiListItemText-inset",
                  "multiline": "MuiListItemText-multiline",
                  "primary": "MuiListItemText-primary",
                  "root": "MuiListItemText-root SimpleListItem-textContainer-2",
                  "secondary": "MuiListItemText-secondary",
                }
              }
            >
              <div
                className="MuiListItemText-root SimpleListItem-textContainer-2"
              >
                <WithStyles(ForwardRef(Typography))
                  className="MuiListItemText-primary"
                  component="span"
                  display="block"
                  variant="body1"
                >
                  <ForwardRef(Typography)
                    className="MuiListItemText-primary"
                    classes={
                      Object {
                        "alignCenter": "MuiTypography-alignCenter",
                        "alignJustify": "MuiTypography-alignJustify",
                        "alignLeft": "MuiTypography-alignLeft",
                        "alignRight": "MuiTypography-alignRight",
                        "body1": "MuiTypography-body1",
                        "body2": "MuiTypography-body2",
                        "button": "MuiTypography-button",
                        "caption": "MuiTypography-caption",
                        "colorError": "MuiTypography-colorError",
                        "colorInherit": "MuiTypography-colorInherit",
                        "colorPrimary": "MuiTypography-colorPrimary",
                        "colorSecondary": "MuiTypography-colorSecondary",
                        "colorTextPrimary": "MuiTypography-colorTextPrimary",
                        "colorTextSecondary": "MuiTypography-colorTextSecondary",
                        "displayBlock": "MuiTypography-displayBlock",
                        "displayInline": "MuiTypography-displayInline",
                        "gutterBottom": "MuiTypography-gutterBottom",
                        "h1": "MuiTypography-h1",
                        "h2": "MuiTypography-h2",
                        "h3": "MuiTypography-h3",
                        "h4": "MuiTypography-h4",
                        "h5": "MuiTypography-h5",
                        "h6": "MuiTypography-h6",
                        "noWrap": "MuiTypography-noWrap",
                        "overline": "MuiTypography-overline",
                        "paragraph": "MuiTypography-paragraph",
                        "root": "MuiTypography-root",
                        "srOnly": "MuiTypography-srOnly",
                        "subtitle1": "MuiTypography-subtitle1",
                        "subtitle2": "MuiTypography-subtitle2",
                      }
                    }
                    component="span"
                    display="block"
                    variant="body1"
                  >
                    <span
                      className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                    >
                      <WithStyles(ForwardRef(Typography))
                        aria-hidden={true}
                        variant="srOnly"
                      >
                        <ForwardRef(Typography)
                          aria-hidden={true}
                          classes={
                            Object {
                              "alignCenter": "MuiTypography-alignCenter",
                              "alignJustify": "MuiTypography-alignJustify",
                              "alignLeft": "MuiTypography-alignLeft",
                              "alignRight": "MuiTypography-alignRight",
                              "body1": "MuiTypography-body1",
                              "body2": "MuiTypography-body2",
                              "button": "MuiTypography-button",
                              "caption": "MuiTypography-caption",
                              "colorError": "MuiTypography-colorError",
                              "colorInherit": "MuiTypography-colorInherit",
                              "colorPrimary": "MuiTypography-colorPrimary",
                              "colorSecondary": "MuiTypography-colorSecondary",
                              "colorTextPrimary": "MuiTypography-colorTextPrimary",
                              "colorTextSecondary": "MuiTypography-colorTextSecondary",
                              "displayBlock": "MuiTypography-displayBlock",
                              "displayInline": "MuiTypography-displayInline",
                              "gutterBottom": "MuiTypography-gutterBottom",
                              "h1": "MuiTypography-h1",
                              "h2": "MuiTypography-h2",
                              "h3": "MuiTypography-h3",
                              "h4": "MuiTypography-h4",
                              "h5": "MuiTypography-h5",
                              "h6": "MuiTypography-h6",
                              "noWrap": "MuiTypography-noWrap",
                              "overline": "MuiTypography-overline",
                              "paragraph": "MuiTypography-paragraph",
                              "root": "MuiTypography-root",
                              "srOnly": "MuiTypography-srOnly",
                              "subtitle1": "MuiTypography-subtitle1",
                              "subtitle2": "MuiTypography-subtitle2",
                            }
                          }
                          variant="srOnly"
                        >
                          <span
                            aria-hidden={true}
                            className="MuiTypography-root MuiTypography-srOnly"
                          />
                        </ForwardRef(Typography)>
                      </WithStyles(ForwardRef(Typography))>
                      <WithStyles(ForwardRef(Typography))
                        aria-hidden={true}
                        classes={
                          Object {
                            "root": "null ",
                          }
                        }
                        color="inherit"
                        variant="body2"
                      >
                        <ForwardRef(Typography)
                          aria-hidden={true}
                          classes={
                            Object {
                              "alignCenter": "MuiTypography-alignCenter",
                              "alignJustify": "MuiTypography-alignJustify",
                              "alignLeft": "MuiTypography-alignLeft",
                              "alignRight": "MuiTypography-alignRight",
                              "body1": "MuiTypography-body1",
                              "body2": "MuiTypography-body2",
                              "button": "MuiTypography-button",
                              "caption": "MuiTypography-caption",
                              "colorError": "MuiTypography-colorError",
                              "colorInherit": "MuiTypography-colorInherit",
                              "colorPrimary": "MuiTypography-colorPrimary",
                              "colorSecondary": "MuiTypography-colorSecondary",
                              "colorTextPrimary": "MuiTypography-colorTextPrimary",
                              "colorTextSecondary": "MuiTypography-colorTextSecondary",
                              "displayBlock": "MuiTypography-displayBlock",
                              "displayInline": "MuiTypography-displayInline",
                              "gutterBottom": "MuiTypography-gutterBottom",
                              "h1": "MuiTypography-h1",
                              "h2": "MuiTypography-h2",
                              "h3": "MuiTypography-h3",
                              "h4": "MuiTypography-h4",
                              "h5": "MuiTypography-h5",
                              "h6": "MuiTypography-h6",
                              "noWrap": "MuiTypography-noWrap",
                              "overline": "MuiTypography-overline",
                              "paragraph": "MuiTypography-paragraph",
                              "root": "MuiTypography-root null ",
                              "srOnly": "MuiTypography-srOnly",
                              "subtitle1": "MuiTypography-subtitle1",
                              "subtitle2": "MuiTypography-subtitle2",
                            }
                          }
                          color="inherit"
                          variant="body2"
                        >
                          <p
                            aria-hidden={true}
                            className="MuiTypography-root null  MuiTypography-body2 MuiTypography-colorInherit"
                          >
                            Title text
                          </p>
                        </ForwardRef(Typography)>
                      </WithStyles(ForwardRef(Typography))>
                    </span>
                  </ForwardRef(Typography)>
                </WithStyles(ForwardRef(Typography))>
              </div>
            </ForwardRef(ListItemText)>
          </WithStyles(ForwardRef(ListItemText))>
        </li>
      </ForwardRef(ListItem)>
    </WithStyles(ForwardRef(ListItem))>
  </SimpleListItem>
</WithStyles(SimpleListItem)>
`;
